service/client_state.cc

```diff
@@ -93,9 +93,9 @@ future<> service::client_state::has_function_access(const sstring& ks, const sst
 }
 
 future<> service::client_state::has_column_family_access(const sstring& ks,
-                const sstring& cf, auth::permission p, auth::command_desc::type t) const {
+                const sstring& cf, auth::permission p, auth::command_desc::type t, std::optional<bool> is_vector_indexed) const {
```

Comment:
How did we reach a situation where this flag needs to be an optional<bool> - can't it just be a bool?
More on this below.